core/src/com/fourinachamber/fortyfive/game/card/CardTextureManager.kt

```diff
@@ -0,0 +1,171 @@
+        return variant.texture
+    }
+
+    private fun getCardPixmap(data: CardTextureData, card: Card): Promise<Pixmap> {
+        val pixmap = data.cardPixmap
+        pixmap?.let {
+            data.cardPixmap = it
+            return it.asPromise()
+        }
+        val message = ServiceThreadMessage.LoadCardPixmap(card.name)
+        FortyFive.serviceThread.sendMessage(message)
+        message.promise.onResolve { data.cardPixmap = it }
+        statistics.pixmapLoads++
+        return message.promise
+    }
+
+    private fun createVariant(
+        data: CardTextureData,
+        card: Card,
+        cost: Int,
+        damage: Int
+    ): Promise<Texture> {
+        val pixmapPromise = getCardPixmap(data, card).chain { cardPixmap ->
+            val pixmap = Pixmap(cardPixmap.width, cardPixmap.height, Pixmap.Format.RGBA8888)
+            if (!card.actor.font.isResolved) ResourceManager.forceResolve(card.actor.font)
+            val message = ServiceThreadMessage.DrawCardPixmap(
+                pixmap,
+                cardPixmap,
+                card,
+                damage,
+                cost,
+                null,
+                card.actor.font.getOrError()
+            )
+            FortyFive.serviceThread.sendMessage(message)
+            message.promise
+        }
+        val texturePromise = pixmapPromise.chain { pixmap ->
+            FortyFive.mainThreadTask {
+                val texture = Texture(pixmap, true)
+                texture.setFilter(
+                    Texture.TextureFilter.MipMapLinearLinear,
+                    Texture.TextureFilter.MipMapLinearLinear
+                )
+                statistics.textureDraws++
+                texture
+            }
+        }
+        val variant = CardTextureVariant(cost, damage, pixmapPromise, texturePromise, mutableListOf(card))
+        data.variants.add(variant)
+        return texturePromise
+    }
+
+    fun giveTextureBack(card: Card) {
+        cardTextures.forEach { data ->
+            val variant = data.variants.find { card in it.borrowers } ?: return@forEach
+            variant.borrowers.remove(card)
+            if (variant.borrowers.size > 0) return
+            disposeVariant(variant, data)
+        }
+    }
+
+    private fun disposeVariant(
+        variant: CardTextureVariant,
+        data: CardTextureData
+    ) {
+        val preventCompleteUnload = preventUnloadingOfCard(data)
+        if (preventCompleteUnload && data.isStandardVariant(variant)) return
+        variant.pixmap.onResolve { it.dispose() }
+        variant.texture.onResolve { it.dispose() }
+        data.variants.remove(variant)
+        if (data.variants.isNotEmpty()) return
+        if (preventCompleteUnload) return
+        data.cardPixmap?.dispose()
+        data.cardPixmap = null
+    }
+
+    private fun preventUnloadingOfCard(data: CardTextureData): Boolean =
+        FortyFive.currentGame != null && data.cardName in SaveState.curDeck.cards
```

Comment:
i wouldn't just prevent the unloading whilst being in a fight, but only while travelling on the map, because otherwise it is loaded and unloaded when going from fight to map to fight (maybe, i am not so sure about this though, because i don't know when you always call "disposeVariant")